File: ecomm-front/src/app/models/order.ts
import CartProduct from './cartProduct';

export default class Order {
    
    private _id : number;
    public get id() : number {
        return this._id;
    }
    public set id(v : number) {
        this._id = v;
    }

    
    private _totalPrice : number;
    public get totalPrice() : number {
        return this._totalPrice;
    }
    public set totalPrice(v : number) {
        this._totalPrice = v;
    }

    
    private _orderDate : string;
    public get orderDate() : string {
        return this._orderDate;
    }
    public set orderDate(v : string) {
        this._orderDate = v;
    }
    
    
    private _products : CartProduct[];
    public get products() : CartProduct[] {
        return this._products;
    }
    public set products(v : CartProduct[]) {
        this._products = v;
    }
    
    constructor(id?: number, totalPrice?: number, orderDate?: string, products?: CartProduct[]){
        this._id = id;
        this._totalPrice = totalPrice;
        this._orderDate = orderDate;
        this._products = products;
    }
}

File: ecomm.services/src/main/java/com/joaoigm/ecomm/services/models/orders/Order.java
package com.joaoigm.ecomm.services.models.orders;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.joaoigm.ecomm.services.models.cart.CartProducts;
import com.joaoigm.ecomm.services.models.user.User;

import javax.persistence.*;
import java.time.LocalDateTime;
import java.util.Date;
import java.util.List;

@Entity
@Table(name = "orders")
public class Order {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;

    @OneToMany(cascade = CascadeType.ALL, mappedBy = "order", targetEntity = OrderProducts.class, fetch = FetchType.LAZY)
    private List<OrderProducts> products;

    @ManyToOne
    private User user;

    private Double totalPrice;

    private LocalDateTime orderDate;

    @JsonCreator
    public Order(List<OrderProducts> products, Double totalPrice) {
        this.products = products;
        this.totalPrice = totalPrice;
    }

    public Order(){}

    public List<OrderProducts> getProducts() {
        return products;
    }

    public void setProducts(List<OrderProducts> products) {
        this.products = products;
    }

    public User getUser() {
        return user;
    }

    public void setUser(User user) {
        this.user = user;
    }

    public Double getTotalPrice() {
        return totalPrice;
    }

    public void setTotalPrice(Double totalPrice) {
        this.totalPrice = totalPrice;
    }

    public LocalDateTime getOrderDate() {
        return orderDate;
    }

    public void setOrderDate(LocalDateTime orderDate) {
        this.orderDate = orderDate;
    }

    public Long getId() {
        return id;
    }
}
